### PR TITLE
added support for installonly packages

### PR DESCRIPTION
**Pull request by kkaarreell**
Allows to create test scenarios like this one:
https://github.com/rpm-software-management/ci-dnf-stack/pull/389

### CODE REVIEW

File: dnf-docker-test/features/steps/rpm_utils.py [hunk 1]
```diff
@@ -120,3 +120,13 @@ def analyze_state(pre, post):
 
     # Most probably we just compare different packages
     assert False, "{!r} -> {!r}".format(pre_nevra, post_nevra)
+
+def is_installonly_pkg(pkg):
+    """
+    Returns True if the package is an installonly package, i.e. provides
+    a package from the defined set.
+    """
+    installonly_pkgs = set([u'kernel', u'kernel-PAE', u'installonlypkg(kernel)', u'installonlypkg(kernel-module)', u'installonlypkg(vm)', u'multiversion(kernel)'])
```
Comment (m-blaha) [inline dnf-docker-test/features/steps/rpm_utils.py]: I'd like to have this set moved out of the function (and thus constructed only once) and named in upper case. Now the set is freshly build in each call of the function.

Comment (m-blaha) [reply dnf-docker-test/features/steps/rpm_utils.py]: and please split this long line to not exceed 100 chars

Comment (kkaarreell) [reply dnf-docker-test/features/steps/rpm_utils.py]: OK. Done.

File: dnf-docker-test/features/steps/rpm_utils.py [hunk 2]
```diff
@@ -120,3 +124,12 @@ def analyze_state(pre, post):
 
     # Most probably we just compare different packages
     assert False, "{!r} -> {!r}".format(pre_nevra, post_nevra)
+
+def is_installonly_pkg(pkg):
+    """
+    Returns True if the package is an installonly package, i.e. provides
+    a package from the defined set.
+    """
+    if set(pkg.provides) & INSTALLONLY_PACKAGES:  # non-empty intersection
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/rpm_utils.py]: Make this oneline by using .is* function of set

File: dnf-docker-test/features/steps/rpm_utils.py [hunk 3]
```diff
@@ -8,6 +8,10 @@
 
 import rpm
 
+INSTALLONLY_PACKAGES = set([u'kernel', u'kernel-PAE', u'installonlypkg(kernel)',
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/rpm_utils.py]: Please `foo = {1, 2}` construction

**Pull request merged**